source/types/Case.ts

```diff
@@ -0,0 +1,267 @@
+import { EncryptionDetails } from "./Encryption";
+
+export type ApplicationStatusType =
+  /** Application has been started by applicant, with possibly partial (and encrypted) answers. */
+  | "active:ongoing"
+  /** Application has been received by the provider and is being processed by a case worker. */
+  | "active:processing"
+  /** Application is fully signed by all applicant and is ready to send in. */
+  | "active:signature:completed"
+  /** Application is not fully signed. One or more applicants need to sign this application. */
+  | "active:signature:pending"
+  /** Application has been submitted to the provider and is awaiting processing. */
+  | "active:submitted"
+  /** Application is closed. */
+  | "closed"
+  /** Application has not yet been started by the applicant. */
+  | "notStarted"
+
+  /** VIVA */
+  /** Application has been sent to VIVA and is awaiting action from a case worker. */
+  | "active:submitted:viva"
+  /** Completion ("stickprovskontroll") is required. */
+  | "active:completionRequired:viva"
+  /** Application is fully approved. Calculation/payment info is available. */
+  | "closed:approved:viva"
+  /** Completion ("stickprovskontroll") has been rejected. */
+  | "closed:completionRejected:viva"
+  /** Application is partially approved. Calcuation/payment info is available. */
+  | "closed:partiallyApproved:viva"
+  /** Application has been processed by a case worker and has been rejected. Rejection info is available. */
+  | "closed:rejected:viva"
+  /** Application has not yet been started by the applicant (VIVA-specific). */
+  | "notStarted:viva";
+
+/**
+ * Possible roles for a person on a case.
+ */
+export type PersonRole = "applicant" | "coApplicant" | "children";
+
+/**
+ * Administrator (case worker).
+ */
+export interface Administrator {
+  email: string;
+  name: string;
+  phone?: string;
+  title: string;
+}
+
+/**
+ * Represents a time period (with a start and end timestamp).
+ */
+export interface Period {
+  /**
+   * Unix timestamp in milliseconds for when this period ends.
+   */
+  endDate: number;
+
+  /**
+   * Unix timestamp in milliseconds for when this period starts.
+   */
+  startDate: number;
+}
+
+/**
+ * Details of a case as retrieved from VIVA.
+ */
+export interface VIVACaseDetails {
+  /**
+   * List of case workers associated with this case.
+   */
+  administrators: Administrator[];
+
+  /**
+   * The period this case pertains to.
+   */
+  period: Period;
+
+  /**
+   * Unique ID associated with this case in VIVA.
+   */
```

Comment:
The term 'case' in Viva is what the applicant may be allocated during the time he or she is entered as an applicant in Viva's register. On the 'case', application periods are created called workflows.